packages/@manta-style/helpers/src/types/QueryType.ts

```diff
@@ -1,37 +1,37 @@
-import MantaStyle, {
+import {
   Type,
-  LiteralType,
-  resolveReferencedType,
-} from '@manta-style/runtime';
+  // ArrayLiteral,
+  // Literal,
+  ConstantTypes,
+  Annotation,
+  // resolveReferencedType,
+} from '@manta-style/types';
 
 export default class QueryType extends Type {
   private type: Type;
   constructor(type: Type) {
     super();
     this.type = type;
   }
-  // TODO: Fix typing
-  public deriveLiteral(annotations: any) {
-    const {
-      context: { query },
-    } = MantaStyle;
-    const { type } = resolveReferencedType(this.type);
-    if (type instanceof LiteralType && typeof query === 'object') {
-      // TODO: Fix typings
-      const content = (query as {
-        [key: string]: string | string[] | undefined;
-      })[type.mock()];
-      if (content) {
-        if (typeof content === 'string') {
-          return MantaStyle.Literal(content);
-        } else if (Array.isArray(content)) {
-          // TODO: Expose ArrayLiteral and use it directly
-          return MantaStyle.ArrayLiteral(
-            content.map((item) => MantaStyle.Literal(item)),
-          );
-        }
-      }
-    }
-    return MantaStyle.NeverKeyword;
+  public deriveLiteral(annotations: Annotation[]) {
+    // const {
+    //   context: { query },
+    // } = MantaStyle;
```

Comment:
@tanhauhau any idea to deal with `context`? 😂

Reply:
initially I was thinking of doing `deriveLiteral(annotations, context)`, but then when i actually change the code, i realise there's ton of places to be updated, ie, the entire manta-runtime is written in `deriveLiteral(annotation)` signature, changing this will need to change everywhere, and some places you just write `deriveLiteral([])` without passing from parent annotations, and i have no idea where to get the context from for such cases. i guess you know the runtime better than i am...

Reply:
the way you mentioned was exactly my initial implementation and i found the problem you mentioned too. 😂that's why i'd like to rename `plugin-system` to `core` so it handles both plugin discovery and context. however, it also means `core` cannot import `types` otherwise circular reference happens. so i'm still not sure whether it's a `no way` or `yes way`.

Reply:
I was just thinking about creating a `@manta-style/context` package. the dependency chain would be `context -> types -> core(plugin-system) -> cli + runtime`

Not sure if i create too many packages, but i guess this is how they relate to each other.

Reply:
oh i deleted my previous comment, will elaborate more later. yeah, `types` has to be separated from `runtime` because if you introduce `plugin-system`, then `runtime` might depend on it. Meanwhile in `plugin-system` if you happen to need to reference some types, it needs to depend on `runtime`. so it's a circular reference.

i feel type classes are a shared resource for other packages, that's why it needs to be a standalone package.

Reply:
yea i noticed it's a circular dependency when you try to explain the dependency chain in the previous comment. Maybe I need to research a bit on how other plugin system manage this kind of dependencies.

Reply:
please check out my comment below. i feel `types`, `context` and `core` could be merged as one package as they all are shared resources.